### PR TITLE
configure global state management

### PR DESCRIPTION
**Pull request by GarrettWeinberg**
- use useContext in place of mobx
- set logged in user name after successful login

### CODE REVIEW

File: src/resources/js/App.tsx [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
   return (
-    <ThemeProvider theme={theme}>
-      <BrowserRouter basename='/main_server/'>
-        <CssBaseline />
-        <Router />
-      </BrowserRouter>
-    </ThemeProvider>
+    <AppContext.Provider value={{ userName, setUserName }}>
+      <ThemeProvider theme={theme}>
+        <BrowserRouter basename='/main_server/'>
```
Comment (pjaudiomv) [inline src/resources/js/App.tsx:15]: is this just temporary, not all servers start with `/main_server/`

Comment (jbraswell) [reply src/resources/js/App.tsx:15]: You can parse the actual basename from `new URL(baseUrl)`. The baseUrl is global, defined in the base laravel template.

Comment (GarrettWeinberg) [reply src/resources/js/App.tsx:15]: we will have to figure out something for that.

Comment (GarrettWeinberg) [reply src/resources/js/App.tsx:15]: but otherwise it will default to `/` so we will need to find what the correct permalink is.

**Pull request merged**